python/cudf/cudf/_lib/reduce.pyx

```diff
@@ -31,7 +31,7 @@ def reduce(reduction_op, Column incol, dtype=None, **kwargs):
 
     col_dtype = incol.dtype
     if reduction_op in ['sum', 'sum_of_squares', 'product']:
-        col_dtype = np.find_common_type([col_dtype], [np.int64])
+        col_dtype = np.find_common_type([col_dtype], [np.uint64])
```

Comment:
Earlier with np.int64, we would end up getting signed int instead of unsigned
```
>>> np.find_common_type([np.uint32], [np.int64])
dtype('int64')
```
but with unit64, it will return unsigned for unsigned types.
```
>>> np.find_common_type([np.int64], [np.uint64])
dtype('int64')
>>> np.find_common_type([np.uint32], [np.uint64])
dtype('uint32')

```